### ISSUE TITLE
Confusion about How2Sign splits vs labels for evaluation

### ISSUE THREAD

**Issue by lucasnhandang** [ZechengLi19/Uni-Sign#34]
Hi, I have a question regarding evaluation on the How2Sign dataset.

From reading the paper & code, I understand that the original code only creates labels for train/test splits (since there was no dev set at this time). However, on the official dataset page (https://how2sign.github.io/#download), the data is now split into train/dev/test, which is a bit confusing.

I downloaded the RGB clips (frontal view) version (31GB train, 1.7GB val, 2.2GB test) and loaded them into my model (using the provided pose files on HuggingFace and merged into a single pose_format folder). During evaluation, I encountered the following error:

[fine_tuning] Building dataloaders...
Dataset test: WARNING - 6 samples missing RGB files
Dataset test: 2349 samples in labels, 2343 samples usable (99.7% coverage)
[DataLoader] test: #total 2343

There might be a mismatch between the dataset splits and the provided label files. What would be the correct way to align the dataset splits and labels to ensure a fair and consistent evaluation? Should I stick to the original train/test split from the paper, or adapt to the new train/dev/test split from the dataset release? Thank you so much!

**Comment by ZechengLi19** [comment]
@lucasnhandang This observation can be explained by a preprocessing step we implemented on the raw How2Sign dataset.

When processing the official unsegmented raw data, we identified multiple invalid instances where the annotated end timestamp of a video segment was smaller than its corresponding start timestamp, indicating clear annotation inconsistencies. To ensure data quality, these erroneous samples were filtered out prior to model training and evaluation. This filtering step accounts for the reduction in the total number of valid samples you observed.

Additionally, we recommend downloading and segmenting the original raw videos by yourself instead of using the pre-segment data for experiments. We have empirically observed that the pre-segment data contains the aforementioned quality issues, which can lead to a significant performance degradation.

We recommend you verify this timestamp inconsistency by inspecting the annotations of the missing samples directly in the original unprocessed dataset.

**Comment by lucasnhandang** [comment]
Thanks for the clarification, I was able to confirm this on my side. 

I found several samples with invalid timestamps (end < start), which in some cases caused out-of-bound errors during data loading. This explains the discrepancy in the number of valid samples after preprocessing.

I’ll proceed with filtering these cases and re-segmenting the raw videos as suggested.
